src/sarathi/llm/call_llm.py

```diff
@@ -2,30 +2,58 @@
 import requests
 
 
-def retrieve_api_key():
-    """Retrieves the OpenAI API key from environment variables.
 
-    This function attempts to retrieve the API key from two possible environment variables:
-    SARATHI_OPENAI_API_KEY and OPENAI_API_KEY. If neither variable is set, it raises
-    a ValueError.
+def get_env_var(var_names, default=None, error_msg=None):
+    """Generic function to retrieve environment variables."""
+    for name in var_names:
+        if name in os.environ:
+            return os.environ[name]
+    if default is not None:
+        print(
+            f"Environment variable(s) {var_names} not found. Using default: {default}"
+        )
+        return default
+    raise ValueError(error_msg or f"Environment variable(s) not found: {var_names}")
+
+
+def retrieve_api_key():
+    """
+    Retrieve the OpenAI API key from environment variables.
 
     Returns:
-        The OpenAI API key as a string.
+        str: The OpenAI API key.
 
     Raises:
-        ValueError: If neither environment variable is found.
+        ValueError: If neither SARATHI_OPENAI_API_KEY nor OPENAI_API_KEY is found.
     """
-    try:
-        return os.environ["SARATHI_OPENAI_API_KEY"]
-    except Exception as e:
-        try:
-            return os.environ["OPENAI_API_KEY"]
-        except Exception as e:
-            raise ValueError(
-                "Exception occured: neither SARATHI_OPENAI_API_KEY nor OPENAI_API_KEY is found"
-            )
+    return get_env_var(
+        ["SARATHI_OPENAI_API_KEY", "OPENAI_API_KEY"],
+        error_msg="Neither SARATHI_OPENAI_API_KEY nor OPENAI_API_KEY is found",
+    )
 
 
+def retrieve_llm_url():
+    """
+    Retrieve the OpenAI API endpoint URL from environment variables.
+
+    Returns:
+        str: The OpenAI API endpoint URL. Defaults to 'https://api.openai.com/v1/chat/completions'
+             if OPENAI_ENDPOINT_URL is not set.
+    """
+    return get_env_var(
+        ["OPENAI_ENDPOINT_URL"], default="https://api.openai.com/v1/chat/completions"
+    )
+
```

Comment:
@abhishek9sharma I think there should be another blank line here according to PEP-8: https://peps.python.org/pep-0008/#blank-lines

Reply:
have fixed this

Reply:
Many thanks.